inst/include/Rcpp/internal/r_coerce.h

```diff
@@ -246,13 +246,13 @@ inline const char* coerce_to_string<REALSXP>(double x){
 template <>
 inline const char* coerce_to_string<INTSXP >(int from) {
 	static char buffer[NB] ;
-    snprintf( buffer, NB, "%*d", integer_width(from), from );
+    snprintf(buffer, NB - 1, "%*d", integer_width(from), from);
     return buffer ;
 }
 template <>
 inline const char* coerce_to_string<RAWSXP >(Rbyte from){
 	static char buff[3];
-    ::sprintf(buff, "%02x", from);
+    snprintf(buff, 2, "%02x", from);
```

Comment:
Do we have a unit test for this one? I'm worried we might be truncating the representation here.

Reply:
I would welcome one.  Can you think of one?

I did run it over nearly 2600 CRAN packages though as the usual reverse depends check...

Reply:
I'll try to put one together later this morning.